File: Level2/오픈채팅방.py
def solution(record):
    result = []
    id_nick = {}
    
    for r in record:
        r = r.split()
        if r[0] == "Enter":
            result.append(r[1] + "/님이 들어왔습니다.")
            id_nick[r[1]] = r[2]
        elif r[0] == "Leave":
            result.append(r[1] + "/님이 나갔습니다.")
        else: #Change
            id_nick[r[1]] = r[2]
            
    for i in range(len(result)):
        r = result[i]
        result[i] = id_nick[r.split("/")[0]] + r.split("/")[1]
    return result

File: Level2/오픈채팅방.js
function solution(record) {
    let result = [];
    let id_nick = {};
    for (let r of record) {
        let r_split = r.split(' ');
        if (r_split[0] === 'Enter') {
            result.push(r_split[1] + '/님이 들어왔습니다.');
            id_nick[r_split[1]] = r_split[2];
        } else if (r_split[0] === 'Leave') {
            result.push(r_split[1] + '/님이 나갔습니다.');
        } else {
            id_nick[r_split[1]] = r_split[2];
        }
    }
    for (let i = 0; i < result.length; i++) {
        let cur = result[i];
        result[i] = id_nick[cur.split('/')[0]] + cur.split('/')[1]
    }
    return result;
}
